native/src/main/java/io/ballerina/stdlib/http/api/HttpDispatcher.java

```diff
@@ -159,15 +159,16 @@ public static InterceptorService findInterceptorService(HTTPInterceptorServicesR
             }
 
             InterceptorService service = servicesOnInterface.get(basePath);
-            setInboundReqProperties(inboundReqMsg, validatedUri, basePath);
+            setInboundReqProperties(inboundReqMsg, validatedUri, rawUri, basePath);
             return service;
         } catch (Exception e) {
             throw HttpUtil.createHttpError(e.getMessage(), HttpErrorType.SERVICE_DISPATCHING_ERROR);
         }
     }
 
-    private static void setInboundReqProperties(HttpCarbonMessage inboundReqMsg, URI requestUri, String basePath) {
-        String subPath = URIUtil.getSubPath(requestUri.getRawPath(), basePath);
+    private static void setInboundReqProperties(HttpCarbonMessage inboundReqMsg, URI requestUri,
+                                                String rawUri, String basePath) {
+        String subPath = URIUtil.getSubPath(rawUri, basePath);
```

Comment:
Passing the raw uri as the sub path is affecting the test cases where we have encoded urls and special characters.

Since the issue is with URI library which treats the double slash as the scheme separator. Can't we just pass a default scheme with the raw uri when creating the URI object? i.e. the following change is suggested :
```java
URI validatedUri = getValidatedURI("http://" + uriWithoutMatrixParams);
```
Since we are not using the scheme of this validatedUri, the value we set here does not matter. If we could get the scheme from the inbound request message, we can use that also.

Reply:
Thank tharmi, Yes, the issue is with the scheme separator. This is a much better solution